### PR TITLE
add Maximum, Minimum, MaxPooling1D keras2 layers

### PR DESCRIPTION
**Pull request by simanw304**

### CODE REVIEW

File: pyzoo/zoo/pipeline/api/keras2/layers/merge.py [hunk 1]
```diff
@@ -14,7 +14,48 @@
...
 #
 import sys
+from zoo.pipeline.api.keras2.base import ZooKeras2Layer
 
 if sys.version >= '3':
     long = int
     unicode = str
+
+
+class Maximum(ZooKeras2Layer):
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/keras2/layers/merge.py]: We still need to add a maximum function here.
```
def maximum(inputs):
    """Functional interface to the `Maximum` layer.

    # Arguments
        inputs: A list of input tensors (at least 2).
      # Returns
        A tensor, the element-wise maximum of the inputs.
    """
    return Maximum()(inputs)
```

Comment (zhichao-li) [reply pyzoo/zoo/pipeline/api/keras2/layers/merge.py]: you can add the docttest in `Maximum` class for `maximum` method.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Maximum.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * a single node (also of the same shape).
+ */
+class Maximum[T: ClassTag](
+    override val inputShape: Shape = null)(implicit ev: TensorNumeric[T])
+  extends Merge[T](layers = null, mode = "max", inputShape = inputShape)
+    with Net {
+}
+
+object Maximum {
+  def apply[@specialized(Float, Double) T: ClassTag](
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Maximum.scala]: add a `maximum` method here as well and also a unittest to conver it.

**Comment by zhichao-li** [comment]
LGTM. let's merge this once pass the jenkins test.

File: pyzoo/zoo/pipeline/api/keras/base.py [hunk 3]
```diff
@@ -45,3 +41,9 @@ def __call__(self, x):
                                                   "connectInputs",
                                                   self,
                                                   to_list(x)))
+
+
+class ZooKerasLayer(ZooKerasCreator, KerasLayer, ZooCallable):
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/keras/base.py]: class ZooKerasLayer(ZooKerasCreator, ZooCallable, KerasLayer):

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/MaxPooling1D.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ * 3D tensor with shape: `(batch_size, downsampled_steps, features)`.
+ *
+ * @param poolSize Size of the region to which max pooling is applied. Integer. Default is 2.
+ * @param strides Factor by which to downscale. Integer, or -1. 2 will halve the input.
+ *               If -1, it will default to poolSize. Default is -1.
+ * @param padding One of `"valid"` or `"same"` (case-insensitive). Default is 'valid'.
+ * @tparam T Numeric type of parameter(e.g. weight, bias). Only support float/double now
+ */
+class MaxPooling1D[T: ClassTag](
+   poolSize: Int = 2,
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/MaxPooling1D.scala]: add `val` for all of the parameters or `override val` is it has been defined in the base class.

**Comment by simanw304** [comment]
jenkins passed http://10.239.47.211:18888/job/ZOO-PR-Validation/635/ 
cc @zhichao-li

**Pull request merged**